server/routes/pinIPFS.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction } from 'express';
+import { DB, sequelize } from '../database';
+import ipfs from '../util/ipfs';
+export const Errors = {
+  NotLoggedIn: 'Not logged in',
+  InvalidAddress: 'Invalid address',
+  NoBlobPresent: 'No JSON blob was input',
+};
+
+const pinIPFS = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!req.user) return next(new Error(Errors.NotLoggedIn));
+  if (!req.body.address_id) return next(new Error(Errors.InvalidAddress));
```

Comment:
If `address_id` is referring to an address, we need to make a Sequelize query and verify that this address actually exists before proceeding, and also that it is owned by the logged in user. Remember that anyone on the internet can hit these routes without going through Commonwealth, so we need to ensure that the route is secure against all possible attacks/spoofs.

Reply:
Done